src/WebJobs.Extensions.DurableTask/IPlatformInformation.cs

```diff
@@ -112,5 +112,11 @@ public interface IPlatformInformation
         /// </summary>
         /// <returns>The application container name.</returns>
         string GetContainerName();
+
+        /// <summary>
+        /// Determines whether the user has opted in to the external PowerShell SDK.
+        /// </summary>
+        /// <returns>True if the user has opted in to the external PowerShell SDK. False otherwise.</returns>
+        bool UsesExternalPowerShellSDK();
```

Comment:
We don't have that notion unfortunately.

I also think I should give more context to the `UsesExternalPowerShellSDK` check, that'll clarify things.

Essentially, to opt into the new PS SDK, users need to set an environment variable. This method checks whether that variable was set. Therefore, it is unavoidably very feature-specific and doesn't correspond to a more general "durable engine" version change.

If we're concerned about this type growing too large, I can always read this environment variable directly in the code path where I need it, instead of adding it as a utility method in this class. I don't feel strongly one way or another.

Reply:
This is fine for now. Maybe down the line we can refactor if this continues to grow.